tools/sdk/ld/eagle.app.v6.common.ld

```diff
@@ -157,6 +157,18 @@ SECTIONS
     *.c.o( EXCLUDE_FILE (umm_malloc.c.o) .literal*, \
            EXCLUDE_FILE (umm_malloc.c.o) .text*)
     *.cpp.o(.literal*, .text*)
+    *.pioenvs\\*\\lib*.a:(EXCLUDE_FILE (umm_malloc.o) .literal*, \
```

Comment:
> You are able to turn if off for specific library. See http://docs.platformio.org/en/latest/librarymanager/config.html#libarchive

That seems to be a library-level switch. Can we apply this switch at framework level?

Also, back to my original question, is it possible to tweak object file names at framework level to make them the same as in Arduino? I.e. .c files get compiled into .c.o, .cpp into .cpp.o and so on? Directory layout doesn't matter much in this case, as the LD script currently has filename-dependent rules only.

> Need to maintain all of them

Well, that's the part of platformio's value proposition, so there's probably no way around that.